lib/listen/adapter/base.rb

```diff
@@ -69,19 +69,11 @@ def start
 
         @started = true
 
-        calling_stack = caller.dup
         Listen::Internals::ThreadPool.add do
-          begin
-            @snapshots.values.each do |snapshot|
-              _timed('Record.build()') { snapshot.record.build }
-            end
-            _run
-          rescue
-            msg = 'run() in thread failed: %s:\n'\
-              ' %s\n\ncalled from:\n %s'
-            _log_exception(msg, calling_stack)
-            raise # for unit tests mostly
```

Comment:
This was clever that they kept the original calling stack and logged that too.
Not so clever that
1. They only did it down here rather than in `ThreadPool.add` (where I moved it).
2. They used `\n` inside a single quoted string!